tests/session/transaction-integration-002.phpt

```diff
@@ -0,0 +1,75 @@
+--TEST--
+MongoDB\Driver\Session::startTransaction() Transient Error Test
+--SKIPIF--
+<?php require __DIR__ . "/../utils/basic-skipif.inc"; ?>
+<?php skip_if_not_clean(DATABASE_NAME, COLLECTION_NAME); ?>
+<?php skip_if_not_replica_set(); ?>
+<?php skip_if_server_version('<', '4.0'); ?>
+--FILE--
+<?php
+require_once __DIR__ . "/../utils/basic.inc";
+
+$manager = new MongoDB\Driver\Manager(URI);
+
+/* Create collections as that can't be (automatically) done in a transaction */
```

Comment:
![](https://media.giphy.com/media/3ov9jFY9wrVu6mE8XS/giphy.gif)

So no implicit creation of collections/databases within a transaction? I wasn't aware of that limitation.

Reply:
I suppose the `create` command doesn't have the same issue with `drop` in that it requires a majority WC, correct?

Reply:
Correct. ``create`` isn't two phase, I believe.

And yes, just like older MySQL's, support for DDL changes in transactions doesn't seem to be included. At least we get an exception instead of silently ignoring it.